File: Python/oomph-data-ws/uf_309_test/ufs/int_omega/ABC288_C.py
# https://atcoder.jp/contests/abc288/tasks/abc288_c

from ....uf_309_lib.ufs.int_omega.unionfind_301 import UnionFind

n, m = map(int, input().split())

r = 0
uf = UnionFind(n + 1)

for j in range(m):
    a, b = map(int, input().split())
    if not uf.union(a, b):
        r = r + 1

print(r)

File: CSharp/Oomph.Data/UF09Test/UFs/v301/Static1/ABC288_C.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using Oomph.Data.UF09Lib.UFs.v301;

namespace UF09Test.UFs.v301.Static1
{
	// Test: https://atcoder.jp/contests/abc288/tasks/abc288_c
	// Test: https://atcoder.jp/contests/abc399/tasks/abc399_c
	class ABC288_C
	{
		static int[] Read() => Array.ConvertAll(Console.ReadLine().Split(), int.Parse);
		static (int, int) Read2() { var a = Read(); return (a[0], a[1]); }
		static void Main() => Console.WriteLine(Solve());
		static object Solve()
		{
			var (n, m) = Read2();
			var es = Array.ConvertAll(new bool[m], _ => Read2());

			var uf = new UnionFind(n + 1);
			foreach (var (a, b) in es)
				uf.Union(a, b);
			return m - n - 1 + uf.GroupsCount;
		}
	}
}
